src/Automation/SnapshotCommand.cs

```diff
@@ -2,43 +2,62 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using Axe.Windows.Actions.Contexts;
+using Axe.Windows.Automation.Data;
 using Axe.Windows.Automation.Resources;
 using Axe.Windows.Core.Bases;
 using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Axe.Windows.Automation
 {
     /// <summary>
-    /// Class to take a snapshot (via SnapshotCommand.Execute).
+    /// Class to take a snapshot (via SnapshotCommand.Execute*).
     /// </summary>
     static class SnapshotCommand
     {
         /// <summary>
-        /// Execute the scan
+        /// Execute the scan synchronously
         /// </summary>
         /// <param name="config">A set of configuration options</param>
         /// <param name="scanTools">A set of tools for writing output files,
         /// creating the expected results format, and finding the target element to scan</param>
-        /// <returns>A SnapshotCommandResult that describes the result of the command</returns>
-        public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools scanTools)
+        /// <param name="cancellationToken">A Cancellation token</param>
+        /// <returns>A set of ScanResults objects that describes the result of the command</returns>
+        public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools scanTools, CancellationToken cancellationToken)
```

Comment:
Can we remove the CancellationToken parameter here, since it will only ever be CancellationToken.None, and just pass that directly to GetScanAllResults on line 34? That way we could also revert the changes in `SnapshotCommandTests`.

Reply:
The CancellationToken here will be provided by the caller of the Async method, so we need to plumb it through